ibm_db2/datadog_checks/ibm_db2/ibm_db2.py

```diff
@@ -571,13 +571,15 @@ def get_connection(self):
             return connection
 
     @classmethod
-    def get_connection_data(cls, db, username, password, host, port, tls_cert):
+    def get_connection_data(cls, db, username, password, host, port, security, tls_cert):
         if host:
             target = 'database={};hostname={};port={};protocol=tcpip;uid={};pwd={}'.format(
                 db, host, port, username, password
             )
             username = ''
             password = ''
+            if security == 'ssl':
+                target = '{};security=ssl;'.format(target)
```

Comment:
So that I understand this correctly, this allows for an ssl connection without a certificate? If so, can you indicate that in the parameter description in the `spec.yaml`? I.e. Are there any other pre-requisites that should be documented for this option to work?

Reply:
With a certificate, but without the need to specify one 😉  For [Db2 on Cloud](https://cloud.ibm.com/docs/Db2onCloud?topic=Db2onCloud-getting-started) on IBM Cloud the required certificate is already bundled into the driver package and is used as a default in case no one is specified. Please refer to [SecurityTransportMode IBM data server driver configuration keyword (ibm.com)](https://www.ibm.com/docs/en/db2/11.1?topic=keywords-security) (DashDB being the old name for "Db2 on Cloud") and [Configuring TLS support in non-Java Db2 clients (ibm.com)](https://www.ibm.com/docs/en/db2/11.5?topic=ctsidc-configuring-tls-support-in-non-java-db2-clients) for docs on that.
We are ourselves using it that way to monitor Db2 on Cloud. Until recently we just added the required parameter `Security=ssl` to the end of the port but a recent change (the added validation in https://github.com/DataDog/integrations-core/pull/8934) broke that. Because our port definition looked like `port: 12345;Security=ssl` where the value is of course not a number 😉 That's why we would like to get the parameter properly integrated.

Reply:
Got it, that makes sense. Thanks for the explanation!